### ISSUE TITLE
White Table

### ISSUE THREAD

**Issue by player1537** [EECSSeniorDesignFa16/research#6]
[Current Document](https://github.com/EECSSeniorDesignFa16/research/blob/master/ideas/WhiteTable.md)

> This project would be a table (coffee-table sized?) which is actually a "smart surface". Userse would be able to use special pens to draw/write on it as if it was a white board. The Table would be able to Save your drawings, so you can restore commonly used things when they're needed.
> 
> This would allow many useful things. Coloring/Doodling; drawing boundaries for a table-top game; jotting down quick notes; leaving reminders for other family member; etc.

Thoughts?

**Comment by player1537** [comment]
I'd be down for an idea like this.

[Raspberry Pi Magic Mirror](https://www.raspberrypi.org/blog/magic-mirror/) is a nice variation of this idea.

[Mini Projector](https://www.amazon.com/Brookstone-MP100-Pocket-Projector-Mobile/dp/B008D6JDS8) is an expensive projector, but there are cheaper ones. We could illuminate from above and/or use a webcam or something for detecting hands or pens or whatever.

Tabletop games are an interesting use-case, but it might be better to do something more industry-oriented. Maybe something like [the latter half of this video](https://www.youtube.com/watch?v=b5C_7T2EhSk)

**Comment by JordanShurmer** [comment]
There's also [this Ted talk](https://www.ted.com/talks/johnny_lee_demos_wii_remote_hacks) about a guy who made an interactive projector using a Wiimote and his own Open Source software. We could do something like that to keep it reasonably cheap for an In Home installation.

Ya, I really like the Industry oriented tables like this that are out there now. There's some pretty amazing things. I think a few casinos and places are starting to use them even. There's not really anything targeting a normal In Home use/price though, so I could see that being useful as well.

Either way we'll have to find someone who wants to be the "Client" for one of our projects :)

**Comment by player1537** [comment]
I'm not sure if @arw12625 will join, but I figured I'd go ahead and chime in for him. He likes this idea and thinks there's a lot of cool things we could do with it. Plus it would have several different hardware components so he could contribute more.

In particular, he's interested in building a large capacitive touch surface that we can use.

**Comment by player1537** [comment]
> Userse would be able to use special pens to draw/write on it as if it was a white board. The Table would be able to Save your drawings, so you can restore commonly used things when they're needed.

I think we could take a slightly different approach to this, and rather than just build a single drawing app, we can build it very generically (like a touch screen for laptops) and then build a couple of applications on top of it. The benefit here is that there already exists multiple drawing apps that work out of the box. Since we're only emulating a mouse, we don't have to deal with the fact that 99% of programs won't work with our board, and can instead focus on making most of those programs work decent out of the box, and then building specialized apps that make use of the extra features of the board.

tl;dr Don't build it for the sole purpose of being a whiteboard. Build it so that it can be used for any (Linux) application, and then specialize applications later (e.g. whiteboard).
